mlrun/model.py

```diff
@@ -774,20 +774,22 @@ def validate_notification_params(self):
 
         notification_class.validate_params(secret_params | params)
 
-    def fill_params_from_secret_params(self):
+    def fill_secret_params_from_project_secret(self):
         """
-        Fill the notification params from the secret_params.
+        Fill the notification secret params from the project secret.
+        We are using this function instead of unmask_secret_params_from_project_secret when we run inside the
+        workflow runner pod that doesn't have access to the k8s secrets (but have access to the project secret)
         """
         secret = self.secret_params.get("secret")
         if secret:
             secret_value = mlrun.get_secret_or_env(secret)
             if secret_value:
                 try:
-                    self.params.update(json.loads(secret_value))
+                    self.secret_params = json.loads(secret_value)
                 except ValueError:
                     logger.warning(
-                        "Failed to load secret value to params",
-                        secret_value=secret_value,
+                        "Failed to load secret",
+                        secret=secret,
                     )
```

Comment:
I think it's best to raise an exception here as this will make the notification fail to send.
It shouldn't cancel the workflow from running though so wrap the `get_notifications` with a `try/except` and in that except log a warning that you failed to load the notification and don't send it.

Reply:
I don't think so, because sometime the webhook is not on the secret params but on the params.

Reply:
Assume you need 100% of the configured params. If you are loading params from a secret and they exist but they are malformed, you cannot assume you have some other params. This is a bad state and you should try sending a notification with only half of its params.